--- Python-2.7.2/Modules/_ctypes/libffi/src/closures.c.orig	2014-01-30 20:38:34.535319786 -0800
+++ Python-2.7.2/Modules/_ctypes/libffi/src/closures.c	2014-01-30 20:40:06.339318551 -0800
@@ -122,6 +122,43 @@
 #include <sys/statfs.h>
 #include <stdlib.h>
 
+//  this needs to be protected by an PLATFORM_SHORT_VERSION = 42 if it works
+#include <stdio.h>
+#if 1
+ssize_t getline(char **lineptr, size_t *n, FILE *stream)
+{
+  char *ptr;
+
+  ptr = fgetln(stream, n);
+
+  if (ptr == NULL) {
+    return -1;
+  }
+
+  /* Free the original ptr */
+  if (*lineptr != NULL) free(*lineptr);
+
+  /* Add one more space for '\0' */
+  size_t len = n[0] + 1;
+
+  /* Update the length */
+  n[0] = len;
+
+  /* Allocate a new buffer */
+  *lineptr = malloc(len);
+
+  /* Copy over the string */
+  memcpy(*lineptr, ptr, len-1);
+
+  /* Write the NULL character */
+  (*lineptr)[len-1] = '\0';
+
+  /* Return the length of the new buffer */
+  return len;
+}
+
+#endif // ideally an android platform but we don't have that -D right now
+
 static int selinux_enabled = -1;
 
 static int
@@ -608,3 +645,10 @@
 
 # endif /* ! FFI_MMAP_EXEC_WRIT */
 #endif /* FFI_CLOSURES */
+
+
+
+
+
+
+
